crates/evm-tracing-events/src/marshalled_opcode.rs

```diff
@@ -0,0 +1,285 @@
+//! Marshalled opcode definition and implementations.
+
+use codec::{Decode, Encode};
+use smallvec::{smallvec, SmallVec};
+use sp_core::sp_std::vec::Vec;
+
+/// Marshalled opcode.
+///
+/// 8 cause the longest is 13 and the epmeric estimate of max length
+/// for the most popular ones in 7-ish.
+#[derive(Default, Debug, Clone, PartialEq, Eq)]
+pub struct MarshalledOpcode(pub SmallVec<[u8; 8]>);
+
+impl From<evm::Opcode> for MarshalledOpcode {
+    fn from(opcode: evm::Opcode) -> Self {
+        MarshalledOpcode(smallvec![opcode.as_u8()])
```

Comment:
Why would we do this? The whole point is to keep a string representation instead of the opcode itself at the API boundary.

Reply:
Again: the goal (of the existing approach in master) to allow runtime to fully control what opcodes exist in the EVM, and to keep the client mostly oblivious to them. So we want to improve on that by just inlining the memory - as in using the smallvec instead of vec; but not to move the logic out of runtime.

Reply:
Yes, that's the idea; string has all the semantics that the consumers of the API need, while the opcode needs to be converted to a string, as I understand